insights/parsers/named_checkconf.py

```diff
@@ -136,51 +131,49 @@ class NamedCheckconf(CommandParser):
     Examples:
         >>> type(named_checkconf)
         <class 'insights.parsers.named_checkconf.NamedCheckconf'>
-        >>> named_checkconf.is_dnssec_disabled
-        False
-        >>> named_checkconf.dnssec_line is None
-        True
-        >>> named_checkconf.disable_algorithms
+        >>> named_checkconf.options['disable-algorithms']
         {'.': ['RSAMD5', 'DSA']}
-        >>> named_checkconf.disable_ds_digests
+        >>> named_checkconf.options['disable-ds-digests']
         {'.': ['GOST']}
+        >>> named_checkconf.options['dnssec-enable']
+        'yes'
     """
 
     def __init__(self, context):
         self.is_dnssec_disabled = False  # dnssec is enabled by default
         self.dnssec_line = None
-        self.disable_algorithms = {}
-        self.disable_ds_digests = {}
+        self.options = {}
         super(NamedCheckconf, self).__init__(context)
 
     def parse_content(self, content):
         if not content:
             raise SkipComponent('No content.')
 
-        full_result = '\n'.join(content)
-
-        match_dnssec = DNSSEC_DISABLED.search(full_result)
-        if match_dnssec:
-            self.is_dnssec_disabled = True
-            self.dnssec_line = match_dnssec.group(0)
-
-        self.disable_algorithms = self.retrieve_disabled(DISABLE_ALGORITHMS, full_result)
-        self.disable_ds_digests = self.retrieve_disabled(DISABLE_DS_DIGESTS, full_result)
-
-    def retrieve_disabled(self, section_regex, source):
-        """
-        Parses 'disable-algorithms' or 'disable_ds_digests' section into a dictionary,
-        where the key is a domain and the value is a list of all algorithms/digests associated with it.
-
-        Attributes:
-            section_regex (string): The regular expression for a given section.
-            source (string): The source in which a given section is searched for.
-        """
-        dict_of_sections = dict()
-        for match_entry in section_regex.finditer(source):
-            entry = match_entry.group(0)
-            # collects all values in quotation marks for given section
-            entry_values = [match_value.group(1) for match_value in INNER_VALLUES.finditer(entry)]
-            dict_of_sections[entry_values[0]] = entry_values[1:]
-
-        return dict_of_sections
+        option_values = None
+        for line in content:
+            if not line.strip():
+                continue
+            items = line.strip().split()
+            if items[0] in OPTIONS_ONE_LINE_NAMES:
+                value = ' '.join(items[1:]).strip('\'";')
+                self.options[items[0]] = value
+                # to be compatible with previous code
+                # it will be deleted later when there is no reference with it
+                # plese use self.options to get data now.
+                if items[0] == 'dnssec-enable' and value == 'no':
+                    self.is_dnssec_disabled = True
+                    self.dnssec_line = line.strip()
```

Comment:
This `line.strip()` was called several times inside this for loop. What about call it once at the start of this for loop and reused it in the later part?

Reply:
@JoySnow  Sure, thanks for the suggestion.